app.rb

```diff
@@ -0,0 +1,102 @@
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+require_relative 'students'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  # Method to list all books
+  def list_books
+    puts 'List of all books:'
+    @books.each do |_book|
+      puts "#{book.title} by {book.author}"
+    end
+  end
+
+  # Method to list all people
+  def list_people
+    puts 'List of all people:'
+    @people.each do |_person|
+      puts "#{person.name} (#{person.class})"
+    end
+  end
+
+  # Method to create a person
+  def create_person
+    puts 'Enter the name of the person:'
+    name = gets.chomp
+    puts 'Is this person a student or teacher? (s/t)'
+    type = gets.chomp.downcase
+
+    case type
+    when 's'
+      @people << Students.new(name)
+      puts 'Student {name} has been created.'
+    when 't'
+      @people << Teacher.new(name)
```

Comment:
- [ ] When I tried to create a Teacher I got the following error due to missing argument in Teacher class so please make sure to fix that 
<img width="843" alt="Screenshot 2023-02-18 at 22 25 58" src="https://user-images.githubusercontent.com/61790587/219900359-e6531a93-20fe-4e35-b849-c2233fa5c837.png">

Reply:
I have since enforced taking user age and specialization